docs/source/_src_user_guide/state_propagation/propagation_setup/acceleration_models/thrust.rst

```diff
@@ -0,0 +1,301 @@
+Thrust Guide
+========================
+This page deals with the inclusion of a thrust force into the dynamical model. Note that when using thrust, it may often be desirable to propagate the mass of the vehicle at the same time (removing burnt propellant for instance).
+Details on how to propagate the mass of a body are given in :ref:`mass_dynamics`. Details on combining translation and mass propagators is given in :ref:`multi_type_dynamics`.
+
+Thrust acceleration methods
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Tudat, the acceleration that a body undergoes due to the addition of thrust can be setup in three different ways:
+
+   - :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_custom_function`: this method lets you specify a custom variable thrust vector as a function, a constant specific impulse, and the frame in which the thrust is defined (default: inertial orientation).
+   - :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_and_isp_from_custom_function`: this method adds the capability of specifying a custom variable specific impulse as a function.
+   - :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_direction_and_magnitude`: this method takes :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustDirectionSettings` and :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustMagnitudeSettings` as inputs. These thrust `direction are discussed further on this page: <#thrust-direction>`_ and `magnitude <#thrust-magnitude>`_. The settings are created using factory functions, in the same was as acceleration settings, environment settings, etc.
+   
+
+An typical representative example on how the thrust acceleration can be set up, using the available :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_direction_and_magnitude` function, is provided below:
+
+   .. tabs::
+
+      .. tab:: Python
+
+         The Tudat(Py) API docs give more details on the :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustDirectionSettings` and :class:`~tudatpy.numerical_simulation.propagation_setup.thrust.ThrustMagnitudeSettings` classes.
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/thrust_accelerations_basic_syntax.py
+            :language: python
+
+      .. tab:: C++
+
+         .. literalinclude:: /_src_snippets/simulation/propagation_setup/thrust/thrust_accelerations_basic_syntax.cpp
+            :language: cp
+
+In the above code snippet, note that we define the thrust acceleration as one that the vehicle exerts on itself.
+
+Thrust direction
+~~~~~~~~~~~~~~~~
+
+Below, the different methods that have been implemented to define the direction of the thrust are outlined. Note that these settings are *only* relevant if you use the :func:`~tudatpy.numerical_simulation.propagation_setup.acceleration.thrust_from_direction_and_magnitude` function.
+
```

Comment:
I'd specify here that all these methods return a `ThrustDirectionSettings` object.